docker-app/qfieldcloud/core/drf_utils.py

```diff
@@ -111,7 +114,7 @@ def remove_invalid_fields(
 
             definition_name, attrs = self._parse_definition(field_name)
             alias = attrs.get("alias", definition_name)
-            query_field_name = self._get_query_field(fields, alias)
+            query_field_name = self._get_query_field(list(fields), alias)
```

Comment:
You better change the type of the `_get_query_field` to accept `Iterable` than explicitly converting to `list` here.

Reply:
Correct, in order to increase the flexibility of the code.